crates/extensions/c8y_mapper_ext/src/compatibility_adapter.rs

```diff
@@ -150,9 +150,12 @@ fn convert_from_old_agent_response(
     if let Ok(Value::Object(response)) = serde_json::from_slice(payload) {
         if let Some(Value::String(cmd_id)) = response.get("id") {
             // The new mapper expects command ids with a specific prefix
-            if let Ok(topic) = Topic::new(&format!(
-                "te/device/main///cmd/{cmd_type}/c8y-mapper-{cmd_id}"
-            )) {
+            let topic_name = if cmd_id.contains("c8y-mapper") {
```

Comment:
I just realized that the following is more appropriate.

```suggestion
            let topic_name = if cmd_id.starts_with("c8y-mapper") {
```